core/shared/src/main/scala/zio/prelude/newtypes/package.scala

```diff
@@ -91,4 +91,30 @@ package object newtypes {
   object FailureOut extends NewtypeF
 
   type FailureOut[+A] = FailureOut.Type[A]
+
+  /**
+   * A newtype representing Right-to-left composition of functors.
+   *
+   * If F[_] and G[_] are both Covariant, then Nested[F, G, *] is also a Covariant.
+   *
+   * If F[_] and G[_] are both IdentityBoth, then Nested[F, G, *] is also an IdentityBoth.
+   *
+   * If F[_] and G[_] are both Traversable, then Nested[F, G, *] is also a Traversable.
```

Comment:
I think it would be nice to add tests for that at CovariantSpec, IdentityBothSpec, TraversableSpec

can copy it from my branch:
https://github.com/Badmoonz/zio-prelude/commit/6df1301ec35870736f92a15e4c4f1f58d51a4fd1#diff-82e9bed52abfcc1cabd8fb86c811dcab611fbd08d2ff81124a90af2fb0a78834R21

Reply:
So I tried to add the ones for `CovariantSpec` & `IdentityBothSpec`, and they led to the same build errors for 2.11 in #449. Any idea on how to make them work?

Reply:
After some research , I've found solution : 

```{
        type T[+A] = Nested[Vector, Option]
        testM("Nested[vector,option]")(checkAllLaws(Traversable)(GenFs.nested(GenF.vector, GenF.option) : GenF[T, Random with Sized] , Gen.anyInt))
       }
```

It looks like scala 2.11 compiler cannot derive type `CapF[_[_[_]]]` from first argument GenFs.nested(GenF.vector, GenF.option) which has type `GenF[(lambda[+A] = Nested[Vector, Option, A)#lambda, Random with Sized]`.
I supposed that's because it has lambda type param, but CovariantSpec has tuple, either gens that also return GenF with lambda type param, and it works well